src/trio/_sync.py

```diff
@@ -18,6 +19,11 @@
     from ._core._parking_lot import ParkingLotStatistics
 
 
+class StalledLockError(Exception):
+    """Raised by :meth:`Lock.acquire` and :meth:`StrictFIFOLock.acquire` if the owner
+    exits, or has previously exited, without releasing the lock."""
```

Comment:
I'm not convinced we need a new exception type here - wouldn't raising `BrokenResourceError` make sense?  If we do go with a new type, I'd at least inherit from that.

Reply:
Yeah this might have been because of my comment. Sorry for not being more explicit: IMO parking lots should raise a subclass of `BrokenResourceError`. This would mean that locks can do something like this (for better error messages):

```py
try:
  await parking_lot.park()
except BrokenParkingLot as e:
  raise BrokenResourceError(f"this lock was held by {e.task} when it exited")  # note specifically that task is accessible by `e.task`
```

Reply:
removed `StalledLockError` and reverted to everything using `BrokenResourceError`, but Lock still reraises. I think that's what the conclusion was?